diff -Nurp a/config.mk b/config.mk
--- a/config.mk	2015-11-08 14:39:37.000000000 -0800
+++ b/config.mk	2017-04-20 23:23:04.986863828 -0700
@@ -21,11 +21,11 @@ FREETYPEINC = /usr/include/freetype2
 FREETYPEINC = ${X11INC}/freetype2
 
 # includes and libs
-INCS = -I${X11INC} -I${FREETYPEINC}
+INCS = -isystem ${X11INC} -isystem ${FREETYPEINC}
 LIBS = -L${X11LIB} -lX11 ${XINERAMALIBS} ${FREETYPELIBS}
 
 # flags
-CPPFLAGS = -D_BSD_SOURCE -D_POSIX_C_SOURCE=2 -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
+CPPFLAGS = -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=2 -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
 #CFLAGS   = -g -std=c99 -pedantic -Wall -O0 ${INCS} ${CPPFLAGS}
 CFLAGS   = -std=c99 -pedantic -Wall -Wno-deprecated-declarations -Os ${INCS} ${CPPFLAGS}
 LDFLAGS  = -s ${LIBS}
diff -Nurp a/dwm.c b/dwm.c
--- a/dwm.c	2017-04-20 23:36:57.647405082 -0700
+++ b/dwm.c	2017-04-20 23:36:25.390973414 -0700
@@ -1469,10 +1469,18 @@ resize(Client *c, int x, int y, int w, i
 
 void
 resizebarwin(Monitor *m) {
-	unsigned int w = m->ww;
-	if (showsystray && m == systraytomon(m))
-		w -= getsystraywidth();
-	XMoveResizeWindow(dpy, m->barwin, m->wx, m->by, w, bh);
+    Client *c;
+
+	for (m = mons; m; m = m->next) {
+		for (c = m->clients; c; c = c->next)
+			if (c->isfullscreen)
+				resizeclient(c, m->mx, m->my, m->mw, m->mh);
+
+		unsigned int w = m->ww;
+		if (showsystray && m == systraytomon(m))
+			w -= getsystraywidth();
+		XMoveResizeWindow(dpy, m->barwin, m->wx, m->by, w, bh);
+	}
 }
 
 void
